django/db/models/fields/files.py

```diff
@@ -374,12 +383,20 @@ def __set__(self, instance, value):
         # Model.__init__().  The ImageField.update_dimension_fields method
         # hooked up to the post_init signal handles the Model.__init__() cases.
         # Assignment happening outside of Model.__init__() will trigger the
-        # update right here.
+        # update using ImageFile._set_dimensions
         if previous_file is not None:
-            self.field.update_dimension_fields(instance, force=True)
+
+            # If the original value passed in was a string, it's usually necessary
+            # to convert it to a file object, here done using the descriptor's
+            # __get__ method.
+            if isinstance(value, str):
+                value = getattr(instance, self.field.attname)
+
+            self.field._set_dimensions(instance, value)
 
 
 class ImageFieldFile(ImageFile, FieldFile):
+
```

Comment:
```suggestion
class ImageFieldFile(ImageFile, FieldFile):
```
Avoid adding formatting changes (we try to keep the PRs as small as possible)

Reply:
I think I have to change it, otherwise CI fails due to black?

Reply:
I think I got this. File is formatted correctly now.